### DISCUSSION TITLE
How do I disable valiadation on `connect` relational models?

### DISCUSSION THREAD

**Discussion by songkeys** [prisma/prisma#13565] [Q&A]
I'm using `connect` to create relational models as described in doc:

```js
const user = await prisma.profile.create({
  data: {
    bio: 'Hello World',
    user: {
      connect: { email: 'alice@prisma.io' },
    },
  },
})
```

I also disabled foreign keys using `previewFeatures = ["referentialIntegrity"]` and `referentialIntegrity = "prisma"`.

But it still give me error when the connect field is not existed:

```
An operation failed because it depends on one or more records that were required but not found.
```

The [doc](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#remarks-15) says

> - connect is available as a nested query when you create a new parent record or update an existing parent record.
> - If the related record does not exist, Prisma Client throws an exception.

Is there any way that I could disable this exception? The case is that I only want this relational models for querying, not when inserting and strictly validating my create input. I.e, something like association in sequelize.

Thank you so much!

**Comment by austincrim** [comment]
Hey @Songkeys 👋,

Is there an existing `User` record that has an `email` of `alice@prisma.io`? The `connect` option is used to relate one record to another, so in that case, the connected record **has** to exist. There is a [`connectOrCreate`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#connectorcreate) option you could look into as well.

`referentialIntegrity = "prisma` doesn't actually disable foreign keys, it uses Prisma's internal logic to enforce foreign key constraints ([docs](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-integrity#setting-the-referential-integrity-in-the-datasource)). If you don't want to use or enforce foreign keys, don't use relation fields in your Prisma schema: 

```prisma
model User {
  id        Int     @id @default(autoincrement())
  email     String?
  profileId Int
}

model Profile {
  id     Int @id @default(autoincrement())
  userId Int
}
```

Going this route gives you full control, but there are no guarantees about referential integrity and you must manually enforce any data constraints in your application code.

**Reply by songkeys** [reply]
Hi, @austincrim ,

Thank you so much for this reply.

> If you don't want to use or enforce foreign keys, don't use relation fields in your Prisma schema

But what if I want to query with an `include` option to **join two tables**? I think what I want is like a [`constraints: false` option in sequelize](https://sequelize.org/docs/v6/other-topics/constraints-and-circularities/):

> ```js
> Document.hasMany(Version);
> Document.belongsTo(Version, {
>   as: 'Current',
>   foreignKey: 'currentVersionId',
>   constraints: false // <--
> });
> ```

With this option, the foreign keys will become *actually* virtual without any validation on inserting or updating.

In prisma, the `referentialIntegrity` has now two options according to the [doc](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-integrity#setting-the-referential-integrity-in-the-datasource). As you said, the option `"prisma"` still enforces foreign key constraints. I think there might be a third option like `"disabled"` to fully disable those constraints?

---

Also, I noticed that I could actually set the `id` instead of connecting the model to bypass this constraint:

```js
const user = await prisma.profile.create({
  data: {
    bio: 'Hello World',
    userId: 42,
  },
})
```

When setting `referentialIntegrity = "prisma"`, this won't give me any constraint error. Is this what I want?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
